src/sentry/integrations/discord/utils/channel.py

```diff
@@ -8,15 +10,20 @@
 
 from . import logger
 
+# Same default timeout as Slack
+DISCORD_DEFAULT_TIMEOUT = 10
+
 
 def validate_channel_id(
     channel_id: str, guild_id: str, integration_id: int | None, guild_name: str | None
-) -> None:
+) -> bool:
```

Comment:
What if you wrote a new function that starts the timer, calls `validate_channel_id`, and returns the bool of whether the timeout was hit? It feels odd to return a boolean about a specific error in this function.

Reply:
I think the other thing here is that in the Slack `get_channel_id_with_timeout` I think you're basing this off of, that one checks the time in a while loop whereas this is just at the end.

Reply:
🤔 actually can't you just check if `e.message == TimeoutError` and raise it?

Reply:
Thanks for the feedback! You're right, returning a value just for a test case is unnecessary. Raising an APITimeoutError directly in the function is much cleaner. I've updated the code based on your suggestion.